src/librustc/hir/print.rs

```diff
@@ -565,16 +565,12 @@ impl<'a> State<'a> {
                 self.s.word(";");
                 self.end(); // end the outer ibox
             }
-            hir::ItemKind::Existential(ref exist) => {
-                self.head(visibility_qualified(&item.vis, "existential type"));
+            hir::ItemKind::OpaqueTy(ref opaque_ty) => {
```

Comment:
Nit: consider refactoring out common parts of `::Ty` and `::OpaqueTy` using a higher order function + lambda for differences in the middle.